### PR TITLE
Programming Example: Data Transfer Transpose

### PR DESCRIPTION
**Pull request by hunhoffe**
This example is based on the [DMA Transpose](https://github.com/Xilinx/mlir-aie/blob/cd33847b76a7c3b9f08ea303421778af8291787a/programming_examples/basic/dma_transpose/aie2.py) example in mlir-aie.

There are two versions, one that uses channels and one that uses `dma_memcpy_nd`

### CODE REVIEW

File: programming_examples/data_transfer_transpose/channel/run_makefile.lit [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// (c) Copyright 2024 Advanced Micro Devices, Inc.
+ // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```
Comment (fifield) [inline programming_examples/data_transfer_transpose/channel/run_makefile.lit:2]: Should be MIT license.

File: programming_examples/data_transfer_transpose/dma/run_makefile.lit [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// (c) Copyright 2024 Advanced Micro Devices, Inc.
+ // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```
Comment (fifield) [inline programming_examples/data_transfer_transpose/dma/run_makefile.lit:2]: MIT license

File: programming_examples/data_transfer_transpose/dma/Makefile
Comment (fifield) [inline programming_examples/data_transfer_transpose/dma/Makefile:1]: needs copyright/license

File: programming_examples/data_transfer_transpose/channel/Makefile
Comment (fifield) [inline programming_examples/data_transfer_transpose/channel/Makefile:1]: needs copyright/license.

**Review by fifield — APPROVED** [review]
Just some copyright nits

**Comment by hunhoffe** [comment]
@fifield thanks for the review!

**Pull request merged**